### PR TITLE
Change source headers to use SPDX license identifiers

### LINKED ISSUE #357: Switch to SPDX code headers
As documented on https://spdx.dev

### PR DESCRIPTION
**Pull request by aperezdc**
SSIA. Note that there might be some cases in which the style checker trips because the intention is to only modify the license lines in file headers, and add the missing ones.

Fixes #357

----

Note that the style checker may complain in some cases and IMO for this patch which only touches file headers I think it would be reasonable to land the patch ignoring its complaints 😉

### CODE REVIEW

File: platform/wayland/os-compatibility.h [hunk 1]
```diff
@@ -21,6 +21,8 @@
  * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
+ *
+ * SPDX-License-Identifier: MIT
```
Comment (psaavedra) [inline platform/wayland/os-compatibility.h:25]: Should we modify imported files from other projects as-is?

File: platform/wayland/os-compatibility.c [hunk 2]
```diff
@@ -21,6 +21,8 @@
  * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
+ *
+ * SPDX-License-Identifier: MIT
```
Comment (psaavedra) [inline platform/wayland/os-compatibility.c:25]: ditto.

**Review by psaavedra — APPROVED** [review]
One doubt about how to proceed with the imported files. Other than that. LGTM. r+.

File: platform/wayland/os-compatibility.h (hunk 1 above)
Comment (aperezdc) [reply platform/wayland/os-compatibility.h:25]: Good point, maybe it's better to leave them be. I'll revert this change before landing.

File: platform/wayland/os-compatibility.c (hunk 2 above)
Comment (aperezdc) [reply platform/wayland/os-compatibility.c:25]: Same here, I'll revert this before landing.

Comment (aperezdc) [reply platform/wayland/os-compatibility.c:25]: I will also revert the change in `platform/gtk4/cog-gtk-settings-{dialog,cell-renderer-variant}.[hc]`, which were picked from the WebKit MiniBrowser.

Comment (aperezdc) [reply platform/wayland/os-compatibility.c:25]: Oh, and `platform/drm/kms.[hc]`, which come from libdrm IIRC.

**Pull request merged**